allennlp/tests/semparse/domain_languages/quarel_language_test.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.semparse.domain_languages.quarel_language import QuaRel
+from allennlp.tests.semparse.domain_languages.domain_language_test import check_productions_match
+from allennlp.semparse import ExecutionError
+
+class QuaRelLanguageTest(AllenNlpTestCase):
+    def setUp(self):
+        super().setUp()
+        self.language = QuaRel()
+
+    def test_constant_logical_form(self):
+        assert self.language.execute('world2').number == 2
+        with pytest.raises(ExecutionError, match='Unrecognized constant'):
+            self.language.execute('"and"')
+
+    def test_infer_quarel(self):
+        assert self.language.execute(('(infer (speed higher world1) (friction higher world1) '
+                                      '(friction lower world1))')) == 1
+
+        assert self.language.execute(('(infer (speed higher world2) (friction higher world1) '
+                                      '(friction lower world1))')) == 0
+
+        # Both answer options are correct.
+        assert self.language.execute(('(infer (speed higher world2) (friction higher world1) '
+                                      '(friction higher world1))')) == -2
+
+        # Neither answer option is correct.
+        assert self.language.execute(('(infer (speed higher world2) (friction higher world2) '
+                                      '(friction higher world2))')) == -1
+
+        assert self.language.logical_form_to_action_sequence(('(infer (speed higher world1) '
+                                                              '(friction higher world1) '
+                                                              '(friction lower world1))')) == \
+                ['@start@ -> int',
+                 'int -> [<QuaRelType,QuaRelType,QuaRelType:int>, QuaRelType, QuaRelType, '
+                 'QuaRelType]',
```

Comment:
This is a case where I'd be just fine going a bit over 100 characters, as long as it's not way over.  Pylint only complains if you go over 115.